### PR TITLE
Adding field descriptions and updating mart descriptions

### PR DESCRIPTION
**Pull request by emiliocampa**
Updating as part of my handover.

There is also a change to the way latest_outcome_sent is defined in ls_api_errors_npq_outcomes as I realised it wasn't looking at the right field.

### CODE REVIEW

**Comment by emiliocampa** [comment]
There are still some fields that I need to fill in/clarify, but it would be great to get this first pass reviewed and pushed through, and then I can spend more time getting definitions for the fields still left.

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 1]
```diff
@@ -9,7 +9,79 @@ config {
...
+        funded_retained1_declaration: "True if the declaration is in an eligible, payable, or paid state.",
+        retained2_declaration_id: "ID of the participant's retained-2 declaration.",
+        retained2_declaration_state: "State of the participant's retained-2 declaration.",
+        retained2_declaration_date: "Date evidence was submitted for participant's retained-2 declaration.",
+        funded_retained2_declaration: "True if the declaration is in an eligible, payable, or paid state.",
+        completed_declaration_id: "ID of the participant's completed declaration.",
+        completed_declaration_state: "State of the participant's completed declaration.",
+        funded_completed_declaration: "True if the declaration is in an eligible, payable, or paid state.",
+        completed_declaration_date: "Date evidence was submitted for participant's completed declaration.",
+        latest_outcome_state: "State of participant's latest outcome for their NPQ. Possible fields: Passed, Failed, Voided.",
```
Comment (consult1815) [inline definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:69]: Worth pointing out how outcomes are joined in on the completed declaration?

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 2]
```diff
@@ -9,7 +9,79 @@ config {
...
+        latest_outcome_date: "Date of completion for the participant's latest outcome.",
+        count_of_fails: "Counts the number of Failed outcome states for a participant.",
+        cohort_id: "ID of cohort.",
+        npq_lead_provider_id: "ID of Lead Provider.",
+        participant_identity_id: "",
```
Comment (consult1815) [inline definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:74]: Might be worth saying what a participant_identity_id is and how it differs from profile id

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 3]
```diff
@@ -9,7 +9,79 @@ config {
...
+        primary_establishment: "Records whether the user's establishment at time of registration was a primary establishment or not. Based on GIAS data.",
+        school_ukprn: "UKPRN of the participant's establishment ",
+        teacher_catchment: "The catchment zone the participant works in within the UK i.e. England, Wales, Northern Ireland etc. International applicants recorded as 'another'.",
+        teacher_catchment_country: "The country the participant works in.",
+        teacher_catchment_iso_country_code: "The ISO code for the country the participant works in.",
+        tsf_primary_eligibility: "TRUE/FALSE for whether a school is eligible for TSF (£200) owing to it being a primary establishment.",
+        tsf_primary_plus_eligibility: "TRUE/ FALSE for whether a school is eligible for the base £200 as well as the additional £600 based on whether they are a primary school and have fewer than 150 pupils. Pupil count based on GIAS data at time of registration.",
+        works_in_nursery: "Indicates whether the participant works in a nursery.",
+        works_in_school: "Indicates whether the participant works in a school. ",
+        latest_user_id: ""
```
Comment (consult1815) [inline definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:84]: Might need to clarify what this is.

File: definitions/marts/looker_studio_marts/ecf/ls_induction_start_date.sqlx [hunk 4]
```diff
@@ -9,7 +9,21 @@ config {
     },
     description: "This mart provides a report that gives the induction start/completion date for each FIP ECT participant ID. Currently only the induction completion date is exposed to Lead Providers over the API. This report will help Lead Providers determine where schedule changes are required.",
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ecf/ls_induction_start_date.sqlx:10]: It's pedantic but can you make it participant profile ID

Comment (consult1815) [reply definitions/marts/looker_studio_marts/ecf/ls_induction_start_date.sqlx:10]: Also, is this based on ecf_inductions?

Comment (consult1815) [reply definitions/marts/looker_studio_marts/ecf/ls_induction_start_date.sqlx:10]: What is the primary table used to create this mart? Is it participant profiles? Just curious if the lead provider information is based on the school in the latest induction record.

File: definitions/marts/looker_studio_marts/ls_declarations_provider_names.sqlx [hunk 5]
```diff
@@ -13,9 +13,28 @@ config {
...
+        declaration_date: "Date evidence was received.",
+        declaration_type: "The name of the stage the declaration has been made against.",
+        delivery_partner_id: "ID of the ECF Delivery Partner associated with the Lead Provider-participant combo.",
+        evidence_held: "Details any evidence provided with the declaration.",
+        mentor_user_id: "User ID of the mentor associated with the declared-against ECT",
+        participant_profile_id: "ID of the participant's profile.",
+        pupil_premium: "Boolean for pupil premium eligibility.",
+        sparsity_uplift: "Boolean for sparsity uplift eligibility.",
+        state: "State of the declaration. Can be submitted, eligible, payable, paid, ineligible, voided, awaiting_clawback, and clawed_back.",
+        superseded_by_id: "NEED PROPER DEFINITION",
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ls_declarations_provider_names.sqlx:30]: Should we get that before you go? (Won't hold up this pull request though)

File: definitions/marts/looker_studio_marts/ls_declarations_provider_names.sqlx [hunk 6]
```diff
@@ -13,9 +13,28 @@ config {
...
+        type: "Says if it belongs to either the ECF or NPQ programme.",
+        user_id: "Used ID of the participant being declared against.",
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ls_declarations_provider_names.sqlx:32]: Just guessing but this is from declarations right?

File: definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx [hunk 7]
```diff
@@ -7,9 +7,23 @@ config {
         partitionBy: "DATE(created_at)",
         clusterBy: ["state", "reason", "programme", "cpd_lp_name"]
     },
-    description: "This mart provides a report that gives the withdrawal/deferall reasons for all participants across NPQ and ECF. This information is provided to users in the CPD Self-Serve Dashboard.",
+    description: "This mart provides a report that gives the current training_status for all participants across NPQ and ECF. This information is provided to users in the CPD Self-Serve Dashboard where it is filtered to show only withdrawn or deferred participants and the reason for their withdrawal/deferall.",
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx:10]: Who are the primary users for this?

File: definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx [hunk 8]
```diff
@@ -7,9 +7,23 @@ config {
...
     columns: {
-        participant_profile_id: "ID of each ECT's profile"
+        id: "Unique ID of the record being pulled from the participant_profile_states table.",
+        created_at: "Timestamp of when the record being pulled from participant_profile_states was created. Since participants can go through different training statuses, only the most recently created record for each participant is pulled.",
+        updated_at: "Timestamp of when the record being pulled from participant_profile_states was updated.",
+        cpd_lead_provider_id: "ID of the Lead Provider linked to the participant in participant_profile_states.",
+        participant_profile_id: "ID of each ECT's profile",
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx:16]: Is this just ECTs?

File: definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx [hunk 9]
```diff
@@ -7,9 +7,23 @@ config {
...
+        reason: "Explains why a participant is in a certain state.",
+        state: "This is the participant's training_status. It can be active, deferred, or withdrawn.",
+        last_streamed_event_occurred_at: "Timestamp of when the record  from participant_profile_states was most recently imported or updated",
+        last_streamed_event_type: "Whether the most recent event for this record from participant_profile_states was it being imported or updated.",
+        cohort: "Participant's cohort from either npq_enrolments or ecf_indcutions_dedupe.",
+        schedule_identifier: "Participant's schedule from either npq_enrolments or ecf_indcutions_dedupe.",
+        cpd_lp_name: "Name of the Lead Provider linked to the participant in participant_profile_states.",
+        participant_type: "Details if a participant is an ECT, Mentor, or NPQ.",
+        programme: "Details if a participant is associated with the ECF or NPQ programmes",
+        rn1: "Row number ensuring that we're pulling through the most recently created record for each participant from participant_profile_states."
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ls_participant_profile_states.sqlx:26]: Should we be exposing this field in the mart?

File: definitions/marts/looker_studio_marts/npq/ls_npqeyl_enrolments.sqlx [hunk 10]
```diff
@@ -9,7 +9,85 @@ config {
     },
     description: "This mart provides a report of all Early Years Leadership NPQs (NPQEYL). The table links to fields from GIAS and Ofsted establishments and feeds into the NPQEYL dashboard.",
     columns: {
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/npq/ls_npqeyl_enrolments.sqlx:11]: Is this NPQ enrolments plus some bonus fields?

Comment (consult1815) [reply definitions/marts/looker_studio_marts/npq/ls_npqeyl_enrolments.sqlx:11]: Would recommend updating the fields below for any field you update in NPQ enrolments

**Review by consult1815 — CHANGES_REQUESTED** [review]
I think we need as much clarification as possible on fields and tables so have put some very detailed requests.

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 11]
```diff
@@ -9,7 +9,79 @@ config {
...
+        application_trn: "This is the TRN provided at the point of application.",
+        trn_verified: "",
+        trn_auto_verified: "This refers to whether the TRN provided in the application process was verified.",
+        application_created_at: "Refers to the date the application was created.",
+        headteacher_status: "Indicates whether the participant is a headteacher in their first five years.",
+        eligible_for_funding: "Indicates whether the applicant is eligible for funding for their course.",
+        funding_choice: "If a participant is not eligible for funding, this indicates the source of the funding for their NPQ. Possible fields: Another, School, Self, Trust.",
+        school_urn: "The unique reference number of the participant's school. Private childcare participants do not need a URN.",
+        school_name: "Name of participant's school.",
+        high_pupil_premium: "This indicates whether the uplift payment for high pupil premium is applicable to this participant's started declaration.",
```
[the remark this answers is not in the document]
Comment (emiliocampa) [reply definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:22]: Have added the criteria to the description.

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 12]
```diff
@@ -9,7 +9,79 @@ config {
...
+        la_name: "Name of local authority.",
+        school_postcode: "Postcode of participant's school",
+        course_name: "Text name of NPQ course.",
+        declaration_course_identifier: "Shorthand of NPQ course as stored against the participant's declaration(s).",
+        provider_name: "Name of lead provider.",
+        ecf_user_id: "",
+        targeted_delivery_funding_eligibility: "Indicates whether this NPQ participant is eligible for targeted delivery uplift payment.",
```
[the remark this answers is not in the document]
Comment (emiliocampa) [reply definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:29]: Have added the criteria to the description.

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx [hunk 13]
```diff
@@ -9,7 +9,79 @@ config {
...
+        latest_declaration_date: "Date evidence was submitted for participant's latest declaration.",
+        latest_declaration_updated_at: "Date participant's latest declaration was updated",
+        participant_profile_id: "",
+        application_status: "Indicates whether the application has been approved by the lead provider.Possible fields: Pending, Accepted, Rejected.",
+        updated_at: "Refers to the latest updated date for the application record.",
+        employer_name: "Shows the entered employer name of non-school users. Text field.",
+        employment_role: "Shows the employment role of non-school users. Text field.",
+        employment_type: "For non-school users, shows the type of institution they are employed in. Possible fields: Local authority supply teachers, Local authority virtual school, Young offender institute, Hospital school, Other.",
+        declaration_user_id: "",
+        profile_user_id: "",
```
[the remark this answers is not in the document]
Comment (emiliocampa) [reply definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:49]: Have had a look at the code. profile_user_id is the ID from the participant_profiles table, which we'd usually define as participant_profile_id. We're instead pulling participant_profile_id from declarations instead.

Do the naming conventions need to change here?

File: definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx (hunk 3 above)
Comment (emiliocampa) [reply definitions/marts/bigquery_marts/npq/npq_enrolments.sqlx:84]: If profile_user_id is the ID from the participant_profiles table, which we'd usually define as participant_profile_id, does the CASE statement for this field then need to change?

**Review by consult1815 — APPROVED** [review]
I'm trusting the text changes you've made here.

**Comment by emiliocampa** [comment]
[surprised]     CAMPA, Emilio reacted to your message:
________________________________

**Pull request merged**